In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

In [2]:
chit_data = pd.read_excel("chit fund exercise.xlsx")

In [3]:
chit_data.head()

Month  Contribution  Amount won by the bidder  \
0      1          2000                     40000   
1      2          2000                     42000   
2      3          2000                     45000   
3      4          2000                     48000   
4      5          2000                     40000   

   Chit fund organizer commission  Net amount recd by Bid winner  \
0                            2500                          37500   
1                            2500                          39500   
2                            2500                          42500   
3                            2500                          45500   
4                            2500                          37500   

   Amount returned to everyone in the group  
0                                       400  
1                                       320  
2                                       200  
3                                        80  
4                                       400

In [4]:
print("chit_size:",chit_data.shape)

chit_size: (25, 6)


By data,There are 25 unique chit funders and as per the rules only once the money can be borrowed.first thing totol money calculated by each person in each month is given by

In [5]:
chit_data["total Contribution"] = chit_data["Contribution"] - chit_data["Amount returned to everyone in the group"]


In [6]:
chit_data.head()

Month  Contribution  Amount won by the bidder  \
0      1          2000                     40000   
1      2          2000                     42000   
2      3          2000                     45000   
3      4          2000                     48000   
4      5          2000                     40000   

   Chit fund organizer commission  Net amount recd by Bid winner  \
0                            2500                          37500   
1                            2500                          39500   
2                            2500                          42500   
3                            2500                          45500   
4                            2500                          37500   

   Amount returned to everyone in the group  total Contribution  
0                                       400                1600  
1                                       320                1680  
2                                       200                1800  
3                                        80                1920  
4                                       400                1600

Calculating overall Contribution contributed by each member of the chit fund throughout 25 months. 
overall amount contributed by each member = Sum of (Contribution of every month(2000) - Amount returned) for 25 months

In [7]:
total =chit_data["total Contribution"].sum()
total

43800

Calculating Return/Net Profit obtained by the Bid Winners for every month 
Return = Net amount received by a participant - total amount contributed by a participant


In [8]:
chit_data["return"] = chit_data["Net amount recd by Bid winner"] - total


In [9]:
chit_data.head()

Month  Contribution  Amount won by the bidder  \
0      1          2000                     40000   
1      2          2000                     42000   
2      3          2000                     45000   
3      4          2000                     48000   
4      5          2000                     40000   

   Chit fund organizer commission  Net amount recd by Bid winner  \
0                            2500                          37500   
1                            2500                          39500   
2                            2500                          42500   
3                            2500                          45500   
4                            2500                          37500   

   Amount returned to everyone in the group  total Contribution  return  
0                                       400                1600   -6300  
1                                       320                1680   -4300  
2                                       200                1800   -1300  
3                                        80                1920    1700  
4                                       400                1600   -6300

Calculating Return Percentage obtained by the Bid Winners for every month 

Return percentage = (Amount returned to everyone in the group / Total amount invested) X 100


In [15]:
chit_data["ret%"] = (chit_data["return"]/total)*100


In [16]:
chit_data.head()

Month  Contribution  Amount won by the bidder  \
0      1          2000                     40000   
1      2          2000                     42000   
2      3          2000                     45000   
3      4          2000                     48000   
4      5          2000                     40000   

   Chit fund organizer commission  Net amount recd by Bid winner  \
0                            2500                          37500   
1                            2500                          39500   
2                            2500                          42500   
3                            2500                          45500   
4                            2500                          37500   

   Amount returned to everyone in the group  total Contribution  return  \
0                                       400                1600   -6300   
1                                       320                1680   -4300   
2                                       200                1800   -1300   
3                                        80                1920    1700   
4                                       400                1600   -6300   

        ret%  
0 -14.383562  
1  -9.817352  
2  -2.968037  
3   3.881279  
4 -14.383562

Answers

Q1. What is the Annualized  Return of the person who bids in the last month ?

Formula for Annualized Return which I was able to procure after reading on internet

((1 + return%)^(12/months the chit fund is hold) - 1) * 100 

In [18]:
chit_data.iloc[24]

Month                                          25.000000
Contribution                                 2000.000000
Amount won by the bidder                    50000.000000
Chit fund organizer commission               2500.000000
Net amount recd by Bid winner               47500.000000
Amount returned to everyone in the group        0.000000
total Contribution                           2000.000000
return                                       3700.000000
ret%                                            8.447489
Name: 24, dtype: float64

In [19]:
((((1 + (chit_data["ret%"][24])/100)) ** (12/25)) - 1) * 100

3.969357358648673

Annualized return of the person who bids in last month is: 3.97%

Q2. What is the Annualized Return of the person who bids in the first month ?

In [21]:
chit_data.iloc[0]

Month                                           1.000000
Contribution                                 2000.000000
Amount won by the bidder                    40000.000000
Chit fund organizer commission               2500.000000
Net amount recd by Bid winner               37500.000000
Amount returned to everyone in the group      400.000000
total Contribution                           1600.000000
return                                      -6300.000000
ret%                                          -14.383562
Name: 0, dtype: float64

In [22]:
((((1 + (chit_data["ret%"][0])/100)) ** (12/25)) - 1) * 100

-7.183019602665885

Annualized return of the person who bids in first month is: -7.18%


Q3. Write an Python script which calculates the annualized return of chit fund participant? - Show the Return % for each month's bid winner.


Return percentages of each month's bid winners.


In [23]:
chit_data["ret%"]

0    -14.383562
1     -9.817352
2     -2.968037
3      3.881279
4    -14.383562
5     -9.817352
6     -7.534247
7     -5.251142
8    -14.383562
9     -9.817352
10    -2.968037
11     3.881279
12   -14.383562
13    -9.817352
14    -7.534247
15    -7.534247
16   -12.100457
17    -9.817352
18    -2.968037
19    -2.968037
20    -5.251142
21    -0.684932
22     1.598174
23     3.881279
24     8.447489
Name: ret%, dtype: float64